docs/ilab-model-engine.md

```diff
@@ -0,0 +1,204 @@
+# Redesign ilab to be a Full Fledged Model Engine
+
+This documents describes a new structure for ilab, consisting of sub-commands that act as parent or "group" commands. Currently the only group command is `ilab`. If the models `ilab` produces are meant to be the golden standard for open source AI, the tool to use and manage these models needs to be full fledged and extensible, especially to models we have not produced but should be able to run.
+
+open source container tools like podman an docker are commonly viewed as engines. `ilab` should be a model engine, managing the deployment and creation of AI models. To be clear, I will be using tools like podman as a structural analogy here with full understanding they have different implications and use-cases.
+
+
+## Key Component
+
+### InstructLab Structure Redesign
+
+Here are two outlines. One that represents the state of `ilab` after this enhancement and the second describes `ilab` after I introduce new key commands to this structure. I am including the second outline for clarity purposes, but this EP only covers the contents of the first structure.
+
+```console
+ilab
+|
+|_______model
+|       |
+|       |____convert
+|       |____download
+|       |____train (--convert)
+|       |____serve   (-i)
+|       |____chat
+|       |____inference 
```

Comment:
@khaledsulayman @alimaredia and others feel inference makes more sense. I tend to agree since inference is a more "ai" term than test (from what I know)